lametro/models.py

```diff
@@ -563,3 +563,115 @@ class Meta:
 
     guid = models.CharField(max_length=256)
     name = models.CharField(max_length=256, unique=True)
+
+
+class BillPacket(models.Model):
```

Comment:
Seems like these `Packet` classes have some reusable attributes and logic. What about factoring that out into [Django's flavor of abstract base class](https://docs.djangoproject.com/en/2.2/topics/db/models/#abstract-base-classes)? Something like...

```python
class Packet(models.Model):

    class Meta:
        abstract = True

    updated_at = models.DateTimeField(auto_now=True)
    url = models.URLField()
    ready = models.BooleanField(default=False)

    @property
    def related_entity(self):
        raise NotImplementedError()

    @property
    def related_files(self):
        raise NotImplementedError()

    def save():
        self._merge_docs()
        self.url = settings.MERGER_BASE_URL + '/document/' + getattr(self, self.related_entity).slug
        super().save(*args, **kwargs)

    def is_ready(self):
        if not self.ready:
            response = requests.head(self.url)
            if response.status_code == 200:
                self.ready = True
                super().save()

        return self.ready

     def _merge_docs(self):
        merge_url = settings.MERGER_BASE_URL + '/merge_pdfs/' + getattr(self, self.related_entity).slug
        requests.post(merge_url, json=self.related_files)


class BillPacket(Packet):

    bill = models.OneToOneField(LAMetroBill,
                                related_name='packet',
                                on_delete=models.CASCADE)

    @property
    def related_entity(self):
        return self.bill

    @propert
    def related_files(self):
        ...
```

Reply:
yeah, I went back and forth on this, but I think you are right that this is a good idea.